### PR TITLE
ci: glee docs automation workflow

### LINKED ISSUE #481: Create GitHub Action to Render Docs into AsyncAPI Website Pages
#### Reason/Context
To streamline the documentation process and keep it up-to-date, automating the rendering of documentation from the AsyncAPI specification can be immensely helpful. GitHub Action will be a way to automate repetitive tasks within a GitHub repository for essentially generating documentation from the Glee Docs folder and deploying it to the AsyncAPI website.

#### Description
The action will be triggered on specific events, such as a push to the repository or a pull request to the main branch. The documentation will be rendered and pushed to the appropriate location on the AsyncAPI website to make it accessible to users and developers.

@Souvikns @KhudaDad414 @derberg

### PR DESCRIPTION
**Pull request by AnimeshKumar923**
### Description

This PR adds the necessary workflows required to automate the process of updating Glee docs in the main website.

#### Related issue(s)
Fixes #481

### CODE REVIEW

**Comment by Souvikns** [comment]
@afzal442 can you please review this once?

**Comment by afzal442** [comment]
Thanks @Souvikns for that. But I am not sure on it. Look likes @derberg has reviewed such a gh action workflow before. 😄  sorry @derberg can you approve if it meets the workflow for this event?

File: .github/workflows/update-docs-on-docs-commits.yml
Comment (derberg) [inline .github/workflows/update-docs-on-docs-commits.yml:1]: hey, not my repo, but please consider to not push this workflow here manually but rather push for having a global workflow, as we already have multiple repos that have the same workflow, so better to control it in `.github`

related issue https://github.com/asyncapi/.github/issues/242

File: .github/workflows/update-docs-in-website.yml
Comment (derberg) [inline .github/workflows/update-docs-in-website.yml:1]: the workflow and script looks good.

but did anyone try to first check manually if this is enough? Website docs will not just digest what you send that easily out of the box. I quickly looked into your `docs` dir and noticed:
- you have subfolders
- you have no front matters in markdown files, like title/weights in https://github.com/asyncapi/cli/blob/master/docs/context.md

My recommendation, try to do the same that you do here in the workflow, manually on a website, and see if your docs will render. I would suggest a prerequisite for this PR is to have a draft PR on website that showcases that after merging this PR it will work like a charm

**Review by derberg — COMMENTED** [review]
left 2 comments, hope that helps

File: .github/workflows/update-docs-on-docs-commits.yml
Comment (AnimeshKumar923) [reply .github/workflows/update-docs-on-docs-commits.yml:1]: @derberg should I open a PR with this same file addressing https://github.com/asyncapi/.github/issues/242?

Comment (afzal442) [reply .github/workflows/update-docs-on-docs-commits.yml:1]: I think yes. You should and also revert this file change for [update-docs-on-docs-commits.yml](https://github.com/asyncapi/glee/pull/548/files/fbe02953e6f405a889cb9f319119ed9c756e7305#diff-8eced5d3e71dce429a484047bea8fc9ed6289fc77807c8f9d3ed79df3541b2b1) 🌺

File: .github/workflows/update-docs-in-website.yml
Comment (afzal442) [reply .github/workflows/update-docs-in-website.yml:1]: sure, I will be doing this part. Thanks

File: .github/workflows/update-docs-on-docs-commits.yml
Comment (AnimeshKumar923) [reply .github/workflows/update-docs-on-docs-commits.yml:1]: > I think yes. You should and also revert this file change for [update-docs-on-docs-commits.yml](https://github.com/asyncapi/glee/pull/548/files/fbe02953e6f405a889cb9f319119ed9c756e7305#diff-8eced5d3e71dce429a484047bea8fc9ed6289fc77807c8f9d3ed79df3541b2b1) 🌺

@afzal442 I will revert this. But let's wait for the approval of this idea below from one of the maintainers... 
> open a PR with this same file addressing https://github.com/asyncapi/.github/issues/242?

File: .github/workflows/update-docs-in-website.yml
Comment (AnimeshKumar923) [reply .github/workflows/update-docs-in-website.yml:1]: @afzal442 Thanks! :+1: :rocket:

File: .github/workflows/update-docs-on-docs-commits.yml
Comment (afzal442) [reply .github/workflows/update-docs-on-docs-commits.yml:1]: Hey @AnimeshKumar923 I had already talked to @derberg. He approved it. Plz go ahead

image

Comment (AnimeshKumar923) [reply .github/workflows/update-docs-on-docs-commits.yml:1]: @afzal442 Okay, got it. Will do the needful...

Comment (AnimeshKumar923) [reply .github/workflows/update-docs-on-docs-commits.yml:1]: I guess @Shurtu-gal opened a https://github.com/asyncapi/.github/pull/258 to solve this so I need not open a PR. I'm thinking of waiting for their PR approval.

cc: @derberg

File: .github/workflows/update-docs-in-website.yml
Comment (afzal442) [reply .github/workflows/update-docs-in-website.yml:1]: > you have no front matters in markdown files

sorry @derberg I forgot to tell you about that is you could find those on `/docs/pages/*` folder.

Comment (afzal442) [reply .github/workflows/update-docs-in-website.yml:1]: moreover, this [PR](https://github.com/asyncapi/glee/pull/538) is up to migrate those remaining pages to `/pages/` folder. Hope that makes sense now

File: .github/workflows/update-docs-in-website.yml
Comment (AnimeshKumar923) [reply .github/workflows/update-docs-in-website.yml]: resolved with 4df5c6c

Comment (afzal442) [reply .github/workflows/update-docs-in-website.yml]: Great!

File: .github/workflows/update-docs-on-docs-commits.yml
Comment (afzal442) [reply .github/workflows/update-docs-on-docs-commits.yml:1]: > You should and also revert this file
@AnimeshKumar923 could you also rmv this file? Else, LGTM!

Comment (AnimeshKumar923) [reply .github/workflows/update-docs-on-docs-commits.yml:1]: @afzal442 is it okay to remove? :eyes:

Comment (afzal442) [reply .github/workflows/update-docs-on-docs-commits.yml:1]: > better to control it in .github

so what do you think about it? let me know.

Comment (AnimeshKumar923) [reply .github/workflows/update-docs-on-docs-commits.yml]: resolved with 494d5b2
cc: @afzal442

**Review by afzal442 — APPROVED** [review]
LGTM from my side. 🚀 ... left it to maintainers.

**Comment by AnimeshKumar923** [comment]
> LGTM from my side. 🚀 ... left it to maintainers.

Thank you! :pray:

File: .github/workflows/update-docs-in-website.yml
Comment (KhudaDad414) [inline .github/workflows/update-docs-in-website.yml:8]:
```suggestion
      - 'docs/pages/*.md'
```

**Review by KhudaDad414 — APPROVED** [review]

**Review by KhudaDad414 — APPROVED** [review]

**Comment by KhudaDad414** [comment]
thanks for the contribution @AnimeshKumar923 and @afzal442 . I am merging this PR now.

**Comment by AnimeshKumar923** [comment]
> thanks for the contribution @AnimeshKumar923 and @afzal442 . I am merging this PR now.

@KhudaDad414 Happy to help! :grin:

File: .github/workflows/update-docs-in-website.yml
Comment (AnimeshKumar923) [reply .github/workflows/update-docs-in-website.yml]: Ah, missed this one :sweat_smile:

File: .github/workflows/update-docs-in-website.yml
Comment (derberg) [reply .github/workflows/update-docs-in-website.yml:1]: folks, you can now add `get-global-docs-autoupdate` topic to this repo, and ask @KhudaDad414 to run https://github.com/asyncapi/.github/actions/workflows/global-replicator.yml against this repo

**Pull request merged**
